### ISSUE TITLE
Error if prefer-tacit can't fix, due to function recides in another file.

### ISSUE THREAD

**Issue by ldrick** [jonaskello/eslint-plugin-functional#221]
# Bug Report

Given are two files `number.ts` and `fail.ts`. Probably the rule `prefer-tacit` fails on the marked line in `fail.ts` and cannot resolve. So Eslint in Visual Studio Code throws errors on every edit and save.

```typescript
// number.ts
export const isPositiveInteger = (val: number): boolean => val > 0 && Number.isInteger(val);
```

```typescript
// fail.ts
import * as E from 'fp-ts/lib/Either';
import { pipe } from 'fp-ts/lib/function';
import { isPositiveInteger } from './number';

export const fail = (period = 20): E.Either<string, number> =>
  pipe(
    period,
    E.fromPredicate(
      (p) => isPositiveInteger(p), // this line fails by using prefer-tacit
      () => 'no positive integer given',
    ),
    E.map((p) => p),
  );
```

The same situation is not failing, if combined in `one-file.ts`. Then `prefer-tacit` automatically fixes on save.

```typescript
// one-file.ts
import * as E from 'fp-ts/lib/Either';
import { pipe } from 'fp-ts/lib/function';

const isPositiveInteger = (val: number): boolean => val > 0 && Number.isInteger(val);

export const fail = (period = 20): E.Either<string, number> =>
  pipe(
    period,
    E.fromPredicate(
      (p) => isPositiveInteger(p), // this line will now be automatically rewritten
      () => 'no positive integer given',
    ),
    E.map((p) => p),
  );
```

## Expected behavior

Rewrites the line to: `isPositiveInteger,`

## Actual behavior

It fails, instead of fixing.

## Workaround

Manually rewrite to tacit.

## The Error logged in Visual Studio Code

> [Error - 10:10:51 PM] TypeError: Cannot read property 'type' of undefined
Occurred while linting C:\dev\priv\trade-indicators\src\averages\fail.ts:9
    at isFunctionDeclaration (C:\dev\priv\trade-indicators\.yarn\$$virtual\eslint-plugin-functional-virtual-31265a1762\0\cache\eslint-plugin-functional-npm-3.2.1-e7b0f15ab9-e934d6c391.zip\node_modules\eslint-plugin-functional\lib\index.js:196:17)
    at isFunctionLike (C:\dev\priv\trade-indicators\.yarn\$$virtual\eslint-plugin-functional-virtual-31265a1762\0\cache\eslint-plugin-functional-npm-3.2.1-e7b0f15ab9-e934d6c391.zip\node_modules\eslint-plugin-functional\lib\index.js:203:12)
    at isCallerViolation (C:\dev\priv\trade-indicators\.yarn\$$virtual\eslint-plugin-functional-virtual-31265a1762\0\cache\eslint-plugin-functional-npm-3.2.1-e7b0f15ab9-e934d6c391.zip\node_modules\eslint-plugin-functional\lib\index.js:1582:18)
    at getCallDescriptors (C:\dev\priv\trade-indicators\.yarn\$$virtual\eslint-plugin-functional-virtual-31265a1762\0\cache\eslint-plugin-functional-npm-3.2.1-e7b0f15ab9-e934d6c391.zip\node_modules\eslint-plugin-functional\lib\index.js:1604:37)
    at getDirectCallDescriptors (C:\dev\priv\trade-indicators\.yarn\$$virtual\eslint-plugin-functional-virtual-31265a1762\0\cache\eslint-plugin-functional-npm-3.2.1-e7b0f15ab9-e934d6c391.zip\node_modules\eslint-plugin-functional\lib\index.js:1628:16)
    at checkFunction$2 (C:\dev\priv\trade-indicators\.yarn\$$virtual\eslint-plugin-functional-virtual-31265a1762\0\cache\eslint-plugin-functional-npm-3.2.1-e7b0f15ab9-e934d6c391.zip\node_modules\eslint-plugin-functional\lib\index.js:1650:16)
    at C:\dev\priv\trade-indicators\.yarn\$$virtual\eslint-plugin-functional-virtual-31265a1762\0\cache\eslint-plugin-functional-npm-3.2.1-e7b0f15ab9-e934d6c391.zip\node_modules\eslint-plugin-functional\lib\index.js:490:28
    at C:\dev\priv\trade-indicators\.yarn\cache\eslint-npm-7.30.0-8a8a06f9b4-028048847e.zip\node_modules\eslint\lib\linter\safe-emitter.js:45:58
    at Array.forEach (<anonymous>)
    at Object.emit (C:\dev\priv\trade-indicators\.yarn\cache\eslint-npm-7.30.0-8a8a06f9b4-028048847e.zip\node_modules\eslint\lib\linter\safe-emitter.js:45:38)
